test/lint/lint-spelling.py

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2022 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+"""
+Warn in case of spelling errors.
+Note: Will exit successfully regardless of spelling errors.
+"""
+
+from subprocess import check_output, STDOUT, CalledProcessError
+
+IGNORE_WORDS_FILE = 'test/lint/spelling.ignore-words.txt'
+FILES_ARGS = ['git', 'ls-files', '--', ":(exclude)build-aux/m4/", ":(exclude)contrib/seeds/*.txt", ":(exclude)depends/", ":(exclude)doc/release-notes/", ":(exclude)src/leveldb/", ":(exclude)src/crc32c/", ":(exclude)src/qt/locale/", ":(exclude)src/qt/*.qrc", ":(exclude)src/secp256k1/", ":(exclude)src/minisketch/", ":(exclude)src/univalue/", ":(exclude)contrib/builder-keys/keys.txt", ":(exclude)contrib/guix/patches"]
+
+
+def check_codespell_install():
+    try:
+        check_output(["codespell", "--version"])
+    except FileNotFoundError:
+        print("Skipping spell check linting since codespell is not installed.")
+        exit(0)
+
+
+def main():
+    check_codespell_install()
+
+    files = check_output(FILES_ARGS).decode("utf-8").splitlines()
+    codespell_args = ['codespell', '--check-filenames', '--disable-colors', '--quiet-level=7', '--ignore-words={}'.format(IGNORE_WORDS_FILE)] + files
+
+    try:
+        check_output(codespell_args, stderr=STDOUT)
```

Comment:
Will this print the spelling mistakes?

Reply:
For me it does but testing on more platforms might be a good idea :)

```
$ test/lint/lint-spelling.py
configure.ac:364: overriden ==> overridden
configure.ac:860: overriden ==> overridden
src/wallet/coinselection.h:228: Chooose ==> Choose
^ Warning: codespell identified likely spelling errors. Any false positives? Add them to the list of ignored words in test/lint/spelling.ignore-words.txt
```

Reply:
Yeah, it should. I just missed the `print(e.output.decode` line below. :man_facepalming:

Reply:
nit: I guess an alternative would be to use `subprocess.check_call` and not redirect the output at all or `print()` it?